### DISCUSSION TITLE
Skipping output binding because one or more optional files are missing

### DISCUSSION THREAD

**Discussion by zillurbmb51** [nextflow-io/nextflow#3382] [Q&A]
Hi all,

I have a process which execute a matlab code. The process runs without any error. But the `out_ch.view()` shows nothing and I am not getting any output in the out directory. The process is like:

```
process mostest2 {
publishDir params.out2, mode:'copy'

input:
tuple val(a1),path(t1),path(z1),path(z2) from m2
output:
path("*") optional true into most

script:
"""
cd /cluster/projects/p33/users/mohammadzr/metabolomics/scripts/mostest_code/
matlab -nodisplay -nosplash -r "zmat_orig_file='${z1}';zmat_perm_file='${z2}'; num_eigval_to_keep='${a1}'; out='mostest_out_${a1}.mat'; mostest_mental; exit"
echo ${a1} 'done'
"""
}
most.view()
```
Any Help? Attached is the log file.
[nextflow.log.txt](https://github.com/nextflow-io/nextflow/files/9994726/nextflow.log.txt)

**Comment by jemunro** [comment]
> `cd /cluster/projects/p33/users/mohammadzr/metabolomics/scripts/mostest_code/`

You have changed out of the working directory, so none of the outputs from your script will be in the working directory. In general, you should not `cd` in a Nextflow script.
